### ISSUE TITLE
🐛 Bug Report: Share and reverse share emails not working

### ISSUE THREAD

**Issue by marvinruder** [stonith404/pingvin-share#443]
### 👟 Reproduction steps

1. Configure SMTP (iCloud Mail server, `smtp.mail.me.com:587`) and test via “Send test email”, working fine
2. Invite user, user receives email, working fine
3. Click on Forgot Password, user receives email, working fine
4. Verify that “Enable share email recipients” is activated
5. Drop file, add recipient email, create share. **No email is received** by the recipient
6. Create reverse share and activate “Send email notification”, log out, open reverse share link, drop file, add different recipient email, create share. **No email is received** by either the reverse share account holder or the recipient entered while sharing.

### 👍 Expected behavior

All emails are sent and received by the recipients. If emails are not sent, an error message is logged.

### 👎 Actual Behavior

Share and reverse share emails are not received. No logs indicate an issue while sending an email.

### 🌐 Browser

Safari Version 17.4.1 (19618.1.15.11.14) on macOS

**Comment by stonith404** [comment]
That's strange, I've just tested these both cases and it works for me on the latest version. Are the mails maybe in the spam?

**Comment by marvinruder** [comment]
Just tested again with email addresses from multiple providers – no share-related emails are delivered, not even in the Spam folders. The test email still works. Very strange indeed.

**Comment by Zixim** [comment]
same here re the missing emails.

For me, after using a reverse share, the frontend doesn't indicate that there's a new file.
Yet, checking the uploads folder and I can see the newly created share present.

**Comment by Zixim** [comment]
ugh... turns out I wasn't actually using a reverse share to upload a file, I got the URLs mixed up and was actually sharing as a logged-in user.
I did notice that the username in the SMTP config section got mangled by an extra "@" symbol. The username wrongly became : my_username@@my_domain.meh
No clue where the extra "@" came from.

**Comment by stonith404** [comment]
I can't really do much as I can't reproduce this. Normally, it should throw an error if something went wrong while sending the mail. But the strange part is that sending the mail only doesn't work for shares and reverse shares.

If you find out more, let me know.

**Comment by marvinruder** [comment]
I finally managed to set up all mails so that they can be sent from iCloud SMTP server. The Pingvin server itself behaves fine, I confirmed that by acting as a rudimentary SMTP server myself using `netcat` and talking to Pingvin from there. It appears that iCloud uses some sort of spam filter that gets particular nervous when emails

- contain lots of words like “file”, “share”, link,
- contain URLs (long URLs with paths and identifiers appear to be more suspicious than just FQDNs),
- are sent to the email they are being sent from.

Maybe my server provider’s (Hetzner) IP address range is also not well trusted, but I did not try using a proxy.

It is quite annoying that they apply checks like that even after successfully authenticating via STARTTLS, and silently discarding emails right away instead of quarantining them. For people wanting to use iCloud SMTP as well I recommend to change

- the subject “New files shared with you” to “New Pingvin Share available”,
- the Reverse share message body so that it no longer includes the full share URL but only the instance’s FQDN without a path,
- the sending email address to one different from any user account’s.

This is what is working for me at the moment. I will close the issue as Pingvin is not the misbehaving entity in this setup.
